packages/@aws-cdk/cdk/lib/cfn-resource.ts

```diff
@@ -266,6 +253,16 @@ export class CfnResource extends CfnRefElement {
     return properties;
   }
 
+  /**
+   * Return properties modified after initiation
+   *
+   * Resources that expose mutable properties should override this function to
+   * collect and return the properties object for this resource.
+   */
+  protected get updatedProperites(): { [key: string]: any } {
```

Comment:
Makes me wonder, why not just implement this behavior in `renderProperties()`? we already have a hook for derived classes to be able to manipulate the synthesized output, seems like a natural place...

Reply:
moved and simplified some code on the way

Reply:
combined, let me know what you think.

Reply:
Looks great